src/pages/TeamManagment/index.tsx

```diff
@@ -0,0 +1,202 @@
+        sx={{ color: COLORS.text.dark, fontWeight: FONT.fontWeight.large }}
+      >
+        {t('settings.title')}
+      </Typography>
+
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'flex-end',
+          marginBottom: '20px',
+          height: '38px',
+          gap: 2,
+        }}
+      >
+        <TextField
+          label="Search"
+          variant="outlined"
+          value={searchTerm}
+          onChange={handleSearchChange}
+          placeholder="Search by name, email..."
+          sx={{
+            padding: 0,
+            '& .MuiOutlinedInput-root': {
+              height: '100%',
+            },
+            '& .MuiFormLabel-root': {
+              transform: searchTerm
+                ? 'translate(14px, -9px) scale(0.75)'
+                : 'translate(14px, 9px) scale(1)',
+            },
+            '& .MuiFormLabel-root.MuiInputLabel-root.Mui-focused': {
+              transform: 'translate(14px, -9px) scale(0.75)',
+              maxWidth: 'calc(133% - 32px)',
+            },
+          }}
+        />
+        <Select
+          value={filterByRole}
+          onChange={(e) => handleRoleFilterChange(e.target.value)}
+          displayEmpty
+          variant="outlined"
+          sx={{ color: COLORS.text.dark }}
+        >
+          <MenuItem value="">{t('settings.select.allRoles')}</MenuItem>
+          <MenuItem value="DISPATCHER">
+            {t('settings.select.dispatcher')}
+          </MenuItem>
+          <MenuItem value="DRIVER">{t('settings.select.driver')}</MenuItem>
+        </Select>
+        <UserForm
+          mode="create"
+          fetchUsers={fetchUsers}
+          addUserToList={addUserToList}
+        />
+      </Box>
+
+      <Divider />
+
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          padding: '10px 20px',
+          alignItems: 'center',
+        }}
+      >
+        {columns.map(({ key, label }) => (
+          <Box
+            key={key}
+            sx={{
+              display: 'flex',
+              alignItems: 'center',
+              width: key === 'role' ? '15%' : key === 'name' ? '25%' : '20%',
+            }}
+          >
+            <Typography
+              sx={{
+                color: COLORS.text.dark,
+                fontSize: FONT.fontSize.small,
+              }}
+            >
+              {label}
+            </Typography>
+            <IconButton onClick={() => toggleSortOrder(key)}>
+              {sortOrder[validKeys[key] || key] === 'asc' ? (
+                <KeyboardArrowUpIcon />
+              ) : (
+                <KeyboardArrowDownIcon />
+              )}
+            </IconButton>
+          </Box>
+        ))}
+        <Typography
+          sx={{ color: COLORS.text.dark, fontSize: FONT.fontSize.small }}
+        >
+          {t('settings.columns.actions')}
+        </Typography>
+      </Box>
+
+      <Divider />
+
+      {loading ? (
+        <Loader />
```

Comment:
better practice make loader as wrapper. so you make wrapper and props = isLoading, and inside wrapper component you sert condition or show loader or show {child}